src/client.ts

```diff
@@ -28,12 +28,43 @@ export const defaultConfig = (): NerisApiConfig => {
   }
 };
 
-export const createNerisClient = (config: NerisApiConfig = defaultConfig()) => {
+export type NerisClient = Client<paths> & {
+  fetchGeometry: (url: URL) => Promise<components['schemas']['GeoJsonMultiPolygon']>;
+};
+
+export const createNerisClient = (config: NerisApiConfig = defaultConfig()): NerisClient => {
   const client = createClient<paths>({
     baseUrl: config.baseUrl,
     fetch: config.fetch,
   });
-  client.use(authMiddleware(config));
+  const auth = authMiddleware(config);
+  client.use(auth);
+
+  return {
+    ...client,
+    fetchGeometry: async (url: URL) => {
+      let req = new Request(url);
+
+      if (auth.onRequest) {
```

Comment:
Does this set the authorization header with an access token?

Reply:
correct. It leverages the same function that the middlware uses. It signs into `/token` and puts the bearer on the authorization header.